py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,439 @@
+# -*- coding: utf-8 -*-
+"""
+Display upcoming Google Calendar events.
+
+This module will display information about upcoming Google Calendar events in
+one of two formats which can be toggled with a button press. The
+event's URL may also be opened in a web browser with a button press.
+
+Configuration parameters:
+    auth_token: The path to where the access/refresh token will be saved
+        after successful credential authorization.
+        (default '~/.config/py3status/google_calendar.auth_token')
+    button_open: Opens the URL for the clicked on event in the default
+        web browser.
+        (default 3)
+    button_toggle: The button used to toggle between output formats
```

Comment:
`py3status`'s policy is that users should be able to refresh the module using `middle_click`. When I do that here, it will refresh without fetching the data. You may want to make `button_refresh` here and make it possible.

Reply:
I'll have to figure out how to fix the crash bug, then. With button refresh, refreshing rapidly with repeated button clicks will crash py3status. It has something to do with interrupting the process of getting events with `_get_events()` when the module is refreshed too fast. It doesn't log anything when it happens either :(

Maybe I could keep track of how long it has been since the last event refresh to make sure we don't refresh too fast.